code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -1174,52 +1198,62 @@ pub mod pallet {
 			stake: &mut StakeOf<T>,
 			owner: &T::AccountId,
 			penalize_for_early_unlock: bool,
+			// TODO(benluelo): Remove this parameter, the transfers are from the pool account.
 			keep_alive: bool,
 		) -> Result<(), DispatchError> {
 			for (reward_asset_id, reward) in &mut rewards_pool.rewards {
-				let inflation =
-					stake.reductions.get(reward_asset_id).cloned().unwrap_or_else(Zero::zero);
-				let claim = if rewards_pool.total_shares.is_zero() {
-					Zero::zero()
-				} else {
-					reward
-						.total_rewards
-						.safe_mul(&stake.share)?
-						.safe_div(&rewards_pool.total_shares)?
-						.safe_sub(&inflation)?
-				};
+				let claim = claim_of_stake::<T>(
+					stake,
+					&rewards_pool.total_shares,
+					reward,
+					reward_asset_id,
+				)?;
+
+				let possibly_slashed_claim = if penalize_for_early_unlock {
+					let amount_slashed = stake.lock.unlock_penalty.mul_floor(claim);
 
-				let claim = if penalize_for_early_unlock {
-					let slashed_amount = stake.lock.unlock_penalty.mul_floor(claim);
 					Self::deposit_event(Event::<T>::UnstakeRewardSlashed {
 						owner: owner.clone(),
 						pool_id: rewards_pool.asset_id,
 						fnft_instance_id: stake.fnft_instance_id,
 						reward_asset_id: *reward_asset_id,
-						amount_slashed: slashed_amount,
+						amount_slashed,
 					});
+
+					T::Assets::transfer(
+						*reward_asset_id,
+						&Self::pool_account_id(&stake.reward_pool_id),
+						&T::TreasuryAccount::get(),
+						amount_slashed,
+						keep_alive,
+					)?;
+
 					// SAFETY: slashed_amount is <= claim as is shown above
-					claim.defensive_saturating_sub(slashed_amount)
+					claim.defensive_saturating_sub(amount_slashed)
 				} else {
 					claim
 				};
 
-				let claim = sp_std::cmp::min(
-					claim,
+				// REVIEW(benluelo): Is this check/ calculation necessary?
```

Comment:
`Reward.total_rewards` is independent from the amount actually available in the pool, this should either return a `RewardPotEmpty` error if there aren't enough funds in the pool to unstake/ claim

Reply:
`total_rewards - claimed_rewards` is supposed to track the total amount of rewards that can be claimed by all stakers (regardless of it being a pot or pool account). If that is no longer the case then we have a problem. Erroring out is also ok anyway in case there is not enough funds to satisfy the claim. That would indicate a problem with our logic. Which is a good thing to know.

Reply:
Noted - I've updated the comment and will review it in a future PR